src/mds/Server.cc

```diff
@@ -6099,14 +6106,31 @@ void Server::handle_set_vxattr(const MDRequestRef& mdr, CInode *cur)
     else
       layout = mdcache->default_file_layout;
 
-    rest = name.substr(name.find("layout"));
-    if (check_layout_vxattr(mdr, rest, value, &layout) < 0)
-      return;
+    if (is_rmxattr && name == "ceph.dir.layout") {
```

Comment:
What about:

https://github.com/ceph/ceph/blob/b5bae62c00ee03152d353d367d3b4a009dc4b937/src/mds/Server.cc#L6398-L6405

This change will cause regressions. (Make sure your tests try removing `ceph.dir.layout.pool_namespace` and `ceph.file.layout.pool_namespace`.)

Reply:
The `layout_vxattrs.sh` test already removes those xattrs. Can you elaborate further on this comment?

Reply:
Unless I'm mistaken, this PR changes it so that removing `ceph.dir.layout.pool_namespace` is not longer possible, yes? That's the regression I'm referring to.

Reply:
Thanks for clarifying!

The behavior regarding these xattrs is the same before and after this patch.

Here's what happens in both cases:
```
[mycephfs]$ mkdir dir
[mycephfs]$ getfattr -n ceph.dir.layout.pool_namespace dir
dir: ceph.dir.layout.pool_namespace: No such attribute
[mycephfs]$ setfattr -n ceph.dir.layout.pool_namespace -v dirns dir
[mycephfs]$ getfattr -n ceph.dir.layout.pool_namespace dir
# file: dir
ceph.dir.layout.pool_namespace="dirns"

[mycephfs]$ setfattr -x ceph.dir.layout.pool_namespace dir
[mycephfs]$ getfattr -n ceph.dir.layout.pool_namespace dir
# file: dir
ceph.dir.layout.pool_namespace=""
```

Reply:
I see, thanks for checking.